clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -10686,6 +10686,9 @@ getTripleBasedSYCLPostLinkOpts(const ToolChain &TC, const JobAction &JA,
       (NewOffloadDriver || (!Triple.isNVPTX() && !Triple.isAMDGPU())))
     addArgs(PostLinkArgs, TCArgs, {"-emit-only-kernels-as-entry-points"});
 
+  if (TCArgs.hasArg(options::OPT_shared))
+    addArgs(PostLinkArgs, TCArgs, {"-exclude-external-functions"});
+
```

Comment:
Please move to the `getOtherSYCLPostLinkOpts` function.  It doesn't look like this option is dependent on any triple behaviors.

Reply:
Nit: Think I call it getNonTripleBasedSYCLPostLinkOpts now. :-)

Reply:
Moved change to getNonTripleBasedSYCLPostLinkOpts.